webapp/impl/src/main/java/cloud/piranha/webapp/impl/DefaultWebApplication.java

```diff
@@ -1087,6 +1092,16 @@ public InputStream getResourceAsStream(String location) {
         return resourceManager.getResourceAsStream(location);
     }
 
+    private Stream<String> getResourcePathsImpl(String path) throws IOException, URISyntaxException {
+        URL resource = resourceManager.getResource(path);
```

Comment:
The ResourceManager API exposes a getAllLocations stream that gives you the entire list of resources. You probably want to use it and then filter what actually applies in this case.